From 339c19d1b3a7a5139aed8b59bad755ddd0e518ff Mon Sep 17 00:00:00 2001
From: Mikael Magnusson <mikachu@gmail.com>
Date: Sun, 3 Jun 2012 20:18:48 +0200
Subject: [PATCH] Fix crash on unexpected NET_WM_MOVERESIZE_CANCEL messages

gtk+ 3.4 apparently sends these randomly when you select text in a
GtkEntry. This also fixes bug #5460.

--- openbox-3.5.0.orig/openbox/event.c
+++ openbox-3.5.0/openbox/event.c
@@ -1488,7 +1488,8 @@ static void event_handle_client(ObClient
             }
             else if ((Atom)e->xclient.data.l[2] ==
                      OBT_PROP_ATOM(NET_WM_MOVERESIZE_CANCEL))
-                moveresize_end(TRUE);
+                if (moveresize_client)
+                    moveresize_end(TRUE);
         } else if (msgtype == OBT_PROP_ATOM(NET_MOVERESIZE_WINDOW)) {
             gint ograv, x, y, w, h;
 
